cloud/pkg/edgecontroller/controller/upstream.go

```diff
@@ -356,7 +356,7 @@ func (uc *UpstreamController) updatePodStatus() {
 			case model.UpdateOperation:
 				for _, podStatus := range podStatuses {
 					getPod, err := uc.kubeClient.CoreV1().Pods(namespace).Get(context.Background(), podStatus.Name, metaV1.GetOptions{})
-					if errors.IsNotFound(err) {
+					if (err == nil && getPod.UID != podStatus.UID) || errors.IsNotFound(err) {
```

Comment:
in kubeedge, the message is process asynchronously，when the old pod is deleted from k8s, the message may exists and if the same name pod is created, it will incorrectly updating the state of  new pod